sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/ApproximateUnique.java

```diff
@@ -337,32 +335,36 @@ public LargestUnique(long sampleSize) {
        * Adds a value to the heap, returning whether the value is (large enough
        * to be) in the heap.
        */
-      public boolean add(Long value) {
-        if (heap.contains(value)) {
-          return true;
-        } else if (heap.size() < sampleSize) {
-          heap.add(value);
-          return true;
-        } else if (value > heap.element()) {
-          heap.remove();
+      public boolean add(long value) {
+        if (heap.size() >= sampleSize && value < heap.element()) {
+          return false; // Common case as input size increases.
+        }
+        if (!heap.contains(value)) {
```

Comment:
What's the complexity of `contains`? Is it O(log n) or O(n)?

Reply:
O(n) in java.util.PriorityQueue.